src/connection/Connection.ts

```diff
@@ -254,17 +255,27 @@ export class Connection {
     async dropDatabase(): Promise<void> {
         const queryRunner = this.createQueryRunner();
         try {
-            if (this.driver instanceof SqlServerDriver || this.driver instanceof MysqlDriver || this.driver instanceof AuroraDataApiDriver) {
-                const databases: string[] = this.driver.database ? [this.driver.database] : [];
+            if (this.driver instanceof SqlServerDriver || this.driver instanceof MysqlDriver || this.driver instanceof AuroraDataApiDriver || this.driver instanceof AbstractSqliteDriver) {
+                const databases: string[] = [];
                 this.entityMetadatas.forEach(metadata => {
                     if (metadata.database && databases.indexOf(metadata.database) === -1)
                         databases.push(metadata.database);
                 });
+                if (databases.length === 0 && this.driver.database) {
+                    databases.push(this.driver.database);
+                };
 
-                for (const database of databases) {
-                    await queryRunner.clearDatabase(database);
+                // await queryRunner.clearDatabase();
+                if (databases.length === 0) {
```

Comment:
how this is possible if on line 265 you push the database in case of `databases.length === 0` ?

Reply:
that push also has a conditional with `&& this.driver.database` ?? Maybe it has no effect but thought it better to be safe?

Can remove the `.clearDatabase()` if so..